csrc/executor.cpp

```diff
@@ -965,7 +966,7 @@ at::Tensor allocateOutput(
     const FusionExecutor::GlobalBufferInfo& out_info,
     const AliasInfo& alias_info,
     const c10::Device& device,
-    ExpressionEvaluator& ee) {
+    const ExpressionEvaluator& ee) {
```

Comment:
The evaluate method on a `const ExpressionEvaluator&` is different from that on a `ExpressionEvaluator&`. See https://github.com/NVIDIA/Fuser/blob/8ea9064c147756391d554d1bb736618ecfd9b21f/csrc/expr_evaluator.h#L62 and https://github.com/NVIDIA/Fuser/blob/8ea9064c147756391d554d1bb736618ecfd9b21f/csrc/expr_evaluator.h#L54

IIUC from @zasdfgbnm the former doesn't cache the computed polymorphicvalue (the final one or the intermediate ones), so it can be slower. @zasdfgbnm , what's your opinion on this?

Reply:
Jingyue is right that the `const` version of this method does not cache intermediate or final result. But is caching important here for this specific function? My understanding is, this function only uses EE for ATen evaluating the fusion outputs marked as alias, and these tensors will only be used to return to the user but nothing else?

Reply:
https://github.com/NVIDIA/Fuser/blob/8ea9064c147756391d554d1bb736618ecfd9b21f/csrc/executor.cpp#L1050-L1062 can grow into a pathological case where we can end up evaluating `n` output tensors `O(n^2)` times. It's probably a rare case but kind of hard to diagnose when it shows up as a performance problem. Wdyt?

Reply:
That is a good point! Due to this, I would suggest keeping it as a non-constant reference.

Reply:
Ack, thanks guys. I will change it back to a mutable version.